### PR TITLE
Thrift: refactor the response flags into one enum class

### PR DESCRIPTION
**Pull request by JuniorHsu**
Signed-off-by: kuochunghsu <kuochunghsu@pinterest.com>

Commit Message:
We have too many boolean state for response state which makes the code is hard to read.

Additional Description:
Risk Level: low
Testing: unit
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: source/extensions/filters/network/thrift_proxy/router/upstream_request.h [hunk 1]
```diff
@@ -71,15 +71,20 @@ struct UpstreamRequest : public Tcp::ConnectionPool::Callbacks,
   ProtocolPtr protocol_;
   ThriftObjectPtr upgrade_response_;
 
+  enum class ResponseState : uint8_t {
+    ResponseNonStarted = 0,
+    ResponseStarted = 1,
+    ResponseCompleted = 2,
+    ConnectionReleased = 3
+  };
```
Comment (rgs1) [inline source/extensions/filters/network/thrift_proxy/router/upstream_request.h:79]: nit: I don't think we need to repeat Response on every enum member, so maybe something like:

```
  enum class ResponseState : uint8_t {
    NotStarted = 0,
    Started = 1,
    Completed = 2,
    ConnectionReleased = 3
  };
```

I don't like `NotStarted` that much, maybe `None` is better?

**Review by rgs1 — CHANGES_REQUESTED** [review]
Generally LGTM, one nit.

File: source/extensions/filters/network/thrift_proxy/router/upstream_request.h (hunk 1 above)
Comment (JuniorHsu) [reply source/extensions/filters/network/thrift_proxy/router/upstream_request.h:79]: Fixed. Thanks!

**Review by rgs1 — APPROVED** [review]
Thanks!

**Review by rgs1 — APPROVED** [review]
(don't know if i need to re-approve after i've been assigned)

**Review by zuercher — APPROVED** [review]

**Comment by zuercher** [comment]
Huh. Github is mystifying, but it's merged.

**Pull request merged**